vcd/metadata_openapi_common_test.go

```diff
@@ -0,0 +1,282 @@
+func testOpenApiMetadataEntryCRUD(t *testing.T, resourceTemplate, resourceAddress, datasourceTemplate, datasourceAddress string, extraParams StringMap) {
+	preTestChecks(t)
+	var params = StringMap{
+		"Org":  testConfig.VCD.Org,
+		"Vdc":  testConfig.Nsxt.Vdc,
+		"Name": t.Name(),
+	}
+
+	// This output allows to perform some assertions on the ID inside a TypeSet,
+	// which is impossible to obtain otherwise.
+	outputHcl := `
+		output "metadata_id" {
+           value = tolist(` + resourceAddress + `.metadata_entry)[0].id
+        }
+	`
+	templateWithOutput := resourceTemplate + outputHcl
+
+	for extraParam, extraParamValue := range extraParams {
+		params[extraParam] = extraParamValue
+	}
+	testParamsNotEmpty(t, params)
+
+	params["FuncName"] = t.Name() + "NoMetadata"
+	params["Metadata"] = " "
+	noMetadataHcl := templateFill(resourceTemplate, params)
+	debugPrintf("#[DEBUG] CONFIGURATION NoMetadata: %s", noMetadataHcl)
+
+	params["FuncName"] = t.Name() + "Create"
+	params["Metadata"] = getOpenApiMetadataTestingHcl(1, 1, 1, 1, 2, 1, 1)
+	createHcl := templateFill(templateWithOutput, params)
+	debugPrintf("#[DEBUG] CONFIGURATION Create: %s", createHcl)
+
+	params["FuncName"] = t.Name() + "WithDatasource"
+	withDatasourceHcl := templateFill(datasourceTemplate+"\n# skip-binary-test\n"+templateWithOutput, params)
+	debugPrintf("#[DEBUG] CONFIGURATION WithDatasource: %s", withDatasourceHcl)
+
+	params["FuncName"] = t.Name() + "DeleteOneKey"
+	params["Metadata"] = getOpenApiMetadataTestingHcl(1, 1, 0, 1, 2, 1, 1)
+	deleteOneKeyHcl := templateFill(templateWithOutput, params)
+	debugPrintf("#[DEBUG] CONFIGURATION DeleteOneKey: %s", deleteOneKeyHcl)
+
+	params["FuncName"] = t.Name() + "Update"
+	params["Metadata"] = strings.NewReplacer("stringValue", "stringValueUpdated").Replace(params["Metadata"].(string))
+	updateHcl := templateFill(templateWithOutput, params)
+	debugPrintf("#[DEBUG] CONFIGURATION Update: %s", updateHcl)
+
+	params["FuncName"] = t.Name() + "UpdateForceRecreate"
+	params["Metadata"] = strings.NewReplacer("NumberEntry", "StringEntry").Replace(params["Metadata"].(string))
+	updateForceRecreateHcl := templateFill(templateWithOutput, params)
+	debugPrintf("#[DEBUG] CONFIGURATION UpdateForceRecreate: %s", updateForceRecreateHcl)
+
+	params["FuncName"] = t.Name() + "Delete"
+	params["Metadata"] = " "
+	deleteHcl := templateFill(resourceTemplate, params)
+	debugPrintf("#[DEBUG] CONFIGURATION Delete: %s", deleteHcl)
+
+	params["FuncName"] = t.Name() + "WithDefaults"
+	params["Metadata"] = "metadata_entry {\n\tkey = \"defaultKey\"\nvalue = \"defaultValue\"\n}"
+	withDefaults := templateFill(resourceTemplate, params)
+	debugPrintf("#[DEBUG] CONFIGURATION WithDefaults: %s", withDefaults)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	// This is used to validate that metadata IDs don't change despite of an update/delete.
+	cachedId := testCachedFieldValue{}
```

Comment:
I guess this reflects it better as it is checking for `Value`

```suggestion
	cachedValue := testCachedFieldValue{}
```

Reply:
Actually I'm saving the ID of one metadata entry here (so I can test that IDs don't change with subsequent updates/deletes)